modules/ROOT/pages/functions/dev_guide/nodejs/invoking-nodejs.adoc

```diff
@@ -1,11 +1,45 @@
 // Module included in the following assemblies
 // /functions/user_guide/develop-nodejs.adoc
 
-// [id="invoking-nodejs_{context}"]
-// = Invoking Node.js functions
+[id="invoking-nodejs"]
+= Invoking Node.js functions
 
 Functions can be invoked with a simple HTTP request. When an incoming request is received, functions are invoked with a `context` object as the first parameter.
 
+== Context objects
+
+Functions are invoked by calling a `context` object as the first parameter.
+
+.Example context object
+[source,js]
+----
+function processPurchase(context, data)
+----
+
+This object provides access to the incoming HTTP request information, including the HTTP request method, any query strings or headers sent with the request, the HTTP version, and the request body. If the incoming request is a CloudEvent, the CloudEvent itself will also be found on the context object.
+// this isn't clear what it means to me - isn't it all HTTP? Do we mean the CE data?
+// Add context object reference doc
+
+=== Context object methods
+
+The `context` object has a single method, `cloudEventResponse()` that accepts a data value and returns a CloudEvent.
+
+In a Knative system, if a function deployed as a service is invoked by an event broker sending a CloudEvent, the broker will examine the response. If the response is a CloudEvent, this event will then be handled by the broker.
+
+.Example
+[source,js]
+----
+// Expects to receive a CloudEvent with customer data
+function processCustomer(context, customer) {
+  // process the customer
+  const processed = processCustomer(customer);
+  return context.cloudEventResponse(customer);
+}
+----
+
+== Second parameter
+// better name for what this is
```

Comment:
How about "CloudEvent data", since a second parameter is only provided if the incoming request contains a CloudEvent.

Reply:
Great, thank you!